### DISCUSSION TITLE
Layering actors without Tiled

### DISCUSSION THREAD

**Discussion by SimplyFriday** [excaliburjs/Excalibur#1961] [Q&A]
I'm attempting to write my first game for a game jam, and I want to make something without using plugins or external editors. I managed to make a very simple grid for rendering level actors, but I can't get them to respect the Z-Index. Any help would be greatly appreciated! 

```TypeScript
export class levelBuildingHelper {
    static tileWidth = 50;
    static tileHeight = 50;


    public static createWallTile(scene: Scene, color: Color, xPos: number, yPos: number) {
        this.createTile(scene, color, xPos, yPos, true, 1);
    }

    public static createBackgroundTile(scene: Scene, color: Color, xPos: number, yPos: number) {
        this.createTile(scene, color, xPos, yPos, false, -1);
    }

    private static createTile(scene: Scene, color: Color, xPos: number, yPos: number, hasCollision: boolean, z: number) {
        let a = new Actor({
            scene: scene,
            width: this.tileWidth,
            height: this.tileHeight,
            color: color,
            pos: vec(xPos * this.tileWidth, yPos * this.tileHeight),
//          z:z  <- setting this here causes all tiles to render on top of each other, but collision is normal
        });

        scene.add(a);
        a.setZIndex(z); // this just literally does nothing as far as I can tell

        if (hasCollision) {
            a.body.collider.type = CollisionType.Fixed;
        }
    }
}
```

**Comment by SimplyFriday** [comment]
Turns out I'm a bit of a putz. When I add actual drawings to the Actors, rather than just using plain color placeholders, the z indexing seems to work now.

**Reply by SimplyFriday** [reply]
Turns out I was double wrong, Z index still doesn't seem to work correctly if I add new actors later during a scene. The new actors always render on top of the old ones.

**Reply by SimplyFriday** [reply]
Okay, more info (in case it's helpful):
Setting the z index works fine for all actors added during the scene's onInitialize() function, but anything added later (e.g. via a timer or user action) is draw on top of all of the old stuff. I looked a little into the source but my eyes started glazing over once I got to the b trees in SortedList. Not sure if this is a bug or if I'm just using it wrong XD

**Reply by eonarheim** [reply]
Thanks for the detective work! I've been poking around feels like a bug in excalibur v0.24.5. It should be working the way you want, setting z should work anytime.

I'm not sure why it's doing this I'll see if I can come up with a reason, feels like an order of operation bug. The somewhat good news is it's working on latest main (we've removed the SortedList implementation).
